### PR TITLE
Feature/rnd 1233 update the cpr sdk to be able to create basedocument objects

### PR DESCRIPTION
**Pull request by THOR300**
# Description

Pull request to integrate functionality to convert from flat and passage level data in hugging face to `BaseDocument` objects. 

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [ ] Patch
- [X] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail (integrated soon)
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Before submitting

- [X] I've read and followed all steps in the [Making a pull request](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [X] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [X] If this PR fixes a bug, I've added a test that will fail without my fix.
- [X] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

### CODE REVIEW

File: src/cpr_sdk/models/__init__.py [hunk 1]
```diff
@@ -1275,6 +1361,52 @@ def to_huggingface(
...
+                document_language: list = list(document_language)
+                assert len(document_language) in [0, 1]
+                document_lang_df = document_df[
+                    document_df["languages"] == document_language[0]
+                ]
+
+                # TODO Remove flag as out of scope
+                if from_passage_level:
+                    parser_output = passage_level_df_to_parser_output(document_lang_df)
+                    print(parser_output)
```
Comment (THOR300) [inline src/cpr_sdk/models/__init__.py]: Confirm how we create a dataset of the document model to run through metrics.

File: src/cpr_sdk/models/__init__.py [hunk 2]
```diff
@@ -1389,4 +1527,7 @@ def from_huggingface(
         )
 
         # TODO: validate the result coming from the below method
-        return self._from_huggingface_parquet(huggingface_dataset, limit)  # type: ignore
+        if passage_level_and_flat:
+            return self._from_huggingface_passage_level_flat_parquet(huggingface_dataset)  # type: ignore
```
Comment (THOR300) [inline src/cpr_sdk/models/__init__.py]: add limit

Comment (kdutia) [reply src/cpr_sdk/models/__init__.py]: looks like this comment still needs to be addressed?

**Review by kdutia — APPROVED** [review]
lgtm 🫡

**Pull request merged**